src/eth-utils.js

```diff
@@ -1,5 +1,61 @@
 import { firmwareAPI, HARDENED } from './bitbox02';
 
+const getCoinFromChainId = chainId => {
+    switch(chainId) {
+        case 1:
+            return firmwareAPI.messages.ETHCoin.ETH;
+        case 3:
+            return firmwareAPI.messages.ETHCoin.RopstenETH;
+        case 4:
+        case 42:
+            return firmwareAPI.messages.ETHCoin.RinkebyETH;
+        default:
+            throw new Error('Unsupported network');
+    }
+}
+
+/**
+ * @param pathString keypath in string format e.g. m/44'/1'/0'/0
+ * @returns keypath as array e.g. [2147483692, 2147483649, 2147483648, 0]
+ */
+export const getPathFromString = pathString => {
+    const levels = pathString.toLowerCase().split('/');
+    if (levels[0] !== 'm') throw new Error('Invalid keypath');
+    return levels.flatMap(level => {
+        if (level === 'm' || level === '') {
+            return [];
+        }
+        if (level.substring(level.length - 1) === "'") {
+            level = +level.substring(0, level.length - 1) + HARDENED;
```

Comment:
what if the number before `'` is already >= HARDENED?

Reply:
what about an input like `m/-1'`? that should not work but maybe this is turned into an unhardened high index.